src/app/search/HighlightedText.tsx

```diff
@@ -10,12 +10,14 @@ export default function HighlightedText({
   startIndex: number;
   endIndex: number;
   className: string;
-}) {
+}): JSX.Element {
```

Comment:
to fix the return below you can make this component return `NonNullable<React.ReactNode>` which is the case for this element, although no harm if you prefer to return `<>{text}</>`

Reply:
Ah, that's the solution!

Reply:
Ugh, still doesn't work:

```
'HighlightedText' cannot be used as a JSX component.
  Its return type 'boolean | ReactChild | ReactFragment | ReactPortal' is not a valid JSX element.
    Type 'string' is not assignable to type 'Element | null'.ts(2786)
```

What React will use as an element and what JSX types expect are just totally mismatched. Though I can't help but wonder if that means returning a bare string only works by accident...

Reply:
my take on it is that a "node" is any single fragment, like

in `<span>{var1}</span>` , `var1` is a node. it can be a string, undefined, null, array, etc. but a full official Component must `return` much narrower types